src/xtgeo/cube/_cube_roxapi.py

```diff
@@ -87,16 +87,14 @@ def _roxapi_cube_to_xtgeo(self, rox, rcube):  # pragma: no cover
     if roxhandedness == "right":
         self._yflip = -1
 
-    ilstart = rcube.get_inline(0)
-    xlstart = rcube.get_crossline(0)
-    ilincr, xlincr = rcube.inline_crossline_increment
-
-    self._ilines = np.array(
-        range(ilstart, self._ncol + ilstart, ilincr), dtype=np.int32
-    )
-    self._xlines = np.array(
-        range(xlstart, self._nrow + xlstart, xlincr), dtype=np.int32
-    )
+    il_start = rcube.get_inline(0)
+    xl_start = rcube.get_crossline(0)
+    il_incr, xl_incr = rcube.inline_crossline_increment
+    il_end = (self._ncol) * il_incr + il_start
+    xl_end = (self._nrow) * xl_incr + xl_start
+
+    self._ilines = np.array(range(il_start, il_end, il_incr), dtype=np.int32)
```

Comment:
Do we want to include the end inline also here? This range will skip the il_end, but might be that is desired functionality 🙂

Reply:
Yes, I have considered that 👍